### ISSUE TITLE
Can not open page because of ERR_NAME_NOT_RESOLVED

### ISSUE THREAD

**Issue by piovano** [tauri-apps/tauri#2124]
### **Describe the bug**
I have written a simple program, but it doesn't always work properly. Sometimes the home page will not open and gets an error ERR_NAME_NOT_RESOLVED.

### **Expected behavior**
Show home page properly.

### **Screenshots**
![err_name_not_resolved](https://user-images.githubusercontent.com/8413870/123935228-99adfa80-d9c6-11eb-9ff2-d1d98622328f.JPG)

### **Platform and Versions (required):**

```
Operating System - Windows, version 10.0.19042 X64
Webview2 - 91.0.864.59

Node.js environment
  Node.js - 14.17.0
  @tauri-apps/cli - 1.0.0-beta.5
  @tauri-apps/api - Not installed

Global packages
  npm - 6.14.13
  yarn - 1.22.0

Rust environment
  rustc - 1.53.0
  cargo - 1.53.0

App directory structure
/.cargo
/.icons
/.idea
/node_modules
/src-tauri

App
  tauri.rs - 1.0.0-beta.4
  build-type - build
  CSP - default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self'
  distDir - ["../index.html"]
  devPath - ["../index.html"]
```

**Comment by nothingismagick** [comment]
What  home page do you mean? Can you share the contents of your index.html and main.rs and tauri.conf.json?

**Comment by piovano** [comment]
Home page is the `index.html` file in the `distDir` folder.

index.html
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Welcome to Tauri!</title>
  </head>
  <body>
    <h1>Welcome to Tauri!</h1>
  </body>
</html>
```

main.rs
```
#![cfg_attr(
  all(not(debug_assertions), target_os = "windows"),
  windows_subsystem = "windows"
)]

fn main() {
  tauri::Builder::default()
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

tauri.conf.json
```
{
  "build": {
    "distDir": [
      "../index.html"
    ],
    "devPath": [
      "../index.html"
    ],
    "beforeDevCommand": "",
    "beforeBuildCommand": ""
  },
  "tauri": {
    "bundle": {
      "active": false,
      "targets": "all",
      "identifier": "com.tauri.dev",
      "icon": [
        "../.icons/32x32.png",
        "../.icons/128x128.png",
        "../.icons/128x128@2x.png",
        "../.icons/icon.icns",
        "../.icons/icon.ico"
      ],
      "resources": [],
      "externalBin": [],
      "copyright": "",
      "category": "DeveloperTool",
      "shortDescription": "",
      "longDescription": "",
      "deb": {
        "depends": [],
        "useBootstrapper": false
      },
      "macOS": {
        "frameworks": [],
        "minimumSystemVersion": "",
        "useBootstrapper": false,
        "exceptionDomain": ""
      }
    },
    "allowlist": {
      "all": false
    },
    "windows": [
      {
        "title": "Welcome to Tauri!",
        "width": 800,
        "height": 600,
        "resizable": true,
        "fullscreen": false
      }
    ],
    "security": {
      "csp": "default-src blob: data: filesystem: ws: wss: http: https: tauri: 'unsafe-eval' 'unsafe-inline' 'self'"
    },
    "updater": {
      "active": false
    }
  }
}
```

**Comment by FabianLars** [comment]
Looks like the same problem we talked about on discord. See https://discordapp.com/channels/616186924390023171/731495047962558564/859366062646624256 and above.

In short: Windows only. Your frontend files doesn't really matter (error occurs on repo examples too). Build the app (release or debug mode, doesn't matter) and run the exe multiple times. Sometimes it starts correctly, sometimes it doesn't. Sadly it feels like a 50% chance and not a rare occurance, at least on my end.

**Comment by nothingismagick** [comment]
Thanks @FabianLars - it seems like we've got our work cut out for us here.

**Comment by nothingismagick** [comment]
_didn't mean to close_

**Comment by Laegel** [comment]
Also, I think that this tauri.conf.json is wrong:
```json
    "distDir": [
      "../index.html"
    ],
    "devPath": [
      "../index.html"
    ],
```
That should be the directory containing the `index.html` file, not the file itself (unless we changed that rule and I'm not aware of that).

You may fix it like this (if `../` **does** contain your `index.html` file):
```json
    "distDir": [
      "../"
    ],
    "devPath": [
      "../"
    ],
```

**Comment by nothingismagick** [comment]
I also actually faced this yesterday (I couldn't import files outside of that folder), so, you should put that file in a folder - even if our examples are not doing it. But I don't think that this resolves the underlying issue.

**Comment by wusyong** [comment]
We updated the URL scheme on Windows yesterday. And I did see it break a few times.
But it's working all fine for me today.
I suppose it's the URL registration to webveiw2 that went wrong. But since it's a hack already, I don't think we have much could do.
The best thing to do is still wait for [WebView2Feedback#172](https://github.com/MicrosoftEdge/WebView2Feedback/issues/172).

Could you try some clean rebuild and see if anything works well after this?

**Comment by piovano** [comment]
@Laegel This tauri.conf.json is copied from examples/helloworld. I also tried using directory instead of file, but the problem persists.

@wusyong Ok, I will try it later.

**Comment by lucasfernog** [comment]
We added support to distDir and devPath as arrays of file paths, so the config was correct. It only injects the index.html file though.

**Comment by nothingismagick** [comment]
yes - what i found in that approach though, @lucasfernog - is that I could only import other files like js if I was in a folder, not pointing at index.html

**Comment by lucasfernog** [comment]
Yeah you need to manually add all files or folders if using an array.

**Comment by lucasfernog** [comment]
Closed by https://github.com/tauri-apps/wry/pull/324 (will be available on wry 0.10.4 or 0.11).
